### ISSUE TITLE
Add Regex::captures_iter

### ISSUE THREAD

**Issue by robinst** [fancy-regex/fancy-regex#59]
See https://docs.rs/regex/1.3.9/regex/struct.Regex.html#method.captures_iter

Currently, you will have to call `captures_from_pos` repeatedly to get this. Would be good to add this as API. Other things such as the replace API would also make use of it (#49).

**Comment by zyansheep** [comment]
This would be very cool : )
I'm trying to read terminal input and get list of inputs split by spaces except ignoring spaces inside quotes. regex doesn't work so i'm using fancy-regex.

**Comment by 4ntoine** [comment]
+1
My use [case](https://stackoverflow.com/questions/66113765/how-to-rewrite-regexp-without-positive-look-ahead). Any chance to have it working now?

**Comment by robinst** [comment]
@4ntoine Released in version 0.5.0 now, give it a try.

**Comment by zyansheep** [comment]
Thanks! Works perfectly! Extra Tip: Use `.flatten()` to deal with the `Result<Match>` that the `Matches` iterator returns
```rust
let input: Vec<&str> = regex.find_iter(&line[..]).flatten().map(|m|m.as_str()).collect()
```
